Automattic/wp-calypso
# Enable Server Side rendering for logged-out Plugins section

#### Proposed Changes

More info: pau2Xa-4j7-p2

This builds on top of #66169 and adds Server Side Rendering to the logged-out Plugins section.

Changes for this include:
* prefetched data inside route middleware so they're ready for when the components are rendered
* hydrate the client site with server fetched data
* adjust and harden client-specific functions
* force the Swipe layout for the Category links so they get rendered


#### Testing Instructions




* if you're running locally, make sure to run `yarn install` beforehand
* make sure you're logged out 
* disable JavaScript
* all links should be accessible:

* `/plugins`
<img width="320" alt="Screenshot on 2022-08-22 at 15-00-31" src="https://user-images.githubusercontent.com/2749938/185916387-e64916a4-08d8-4992-86db-1bb58f30e68c.png">

* `/plugins?s=seo`
<img width="320" alt="Screenshot on 2022-08-22 at 14-58-31" src="https://user-images.githubusercontent.com/2749938/185916446-3b1aee3d-86ec-4c03-8f6e-b37ace1d874c.png">

* `/plugins/browse/seo`
<img width="320" alt="Screenshot on 2022-08-22 at 14-58-58" src="https://user-images.githubusercontent.com/2749938/185916467-6c3bbebc-57c0-4273-912f-b4eb08fa22e9.png">

* `/plugins/wordpress-seo-premium`

<img width="320" alt="Screenshot on 2022-08-22 at 16-37-48" src="https://user-images.githubusercontent.com/2749938/185934709-b6400b3c-97eb-49b2-a2fa-bad5c663b79a.png">

* clicking on the `Visit plans` CTA should send you to https://wordpress.com/pricing/


* all the links that should be inaccessible and should show the loading placeholder logo (enabling JavaScript would redirect to log in)
  * `/plugins/manage`
  * `/plugins/upload`
  * `/plugins/setup`
  * `/plugins/active`
  * `/plugins/inactive`
  * `/plugins/updates`
  * `/plugins/wordpress-seo-premium/eligibility`
  * any link which has a `SITE_ID` in it (ex. `/plugins/browse/seo/[SITE_ID]`)

 <img width="320" alt="Screenshot on 2022-08-22 at 15-02-47" src="https://user-images.githubusercontent.com/2749938/185916819-b5eec6e4-6b6f-45df-91c0-f3f97799fd3b.png">
 

* enable JavaScript and log in
* visit the same links
* everything should work as before


#### Pre-merge Checklist






Related to #
